### ISSUE TITLE
Validating permissions not correct

### ISSUE THREAD

**Issue by holtkampjs** [rhysd/actionlint#558]
# Summary

I noticed when taking a peek at [`rule_permissions.go`](https://github.com/rhysd/actionlint/blob/main/rule_permissions.go) that all permissions are validated against `read`, `write`, and `none`.

## Details

Simply checking for `read`, `write`, or `none` is mostly correct but as you can see in the docs, [Defining access for the GITHUB_TOKEN scopes](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#defining-access-for-the-github_token-scopes), the `models` and `id-token` scopes do not support all three of `read`, `write` or `none` permissions.

```yaml
permissions:
  actions: read|write|none
  attestations: read|write|none
  checks: read|write|none
  contents: read|write|none
  deployments: read|write|none
  id-token: write|none
  issues: read|write|none
  models: read|none
  discussions: read|write|none
  packages: read|write|none
  pages: read|write|none
  pull-requests: read|write|none
  security-events: read|write|none
  statuses: read|write|none
```

## Expected behavior

Similar to the output when a scope is misspelled:
```log
.github/workflows/something.yaml:24:13: "rea" is invalid for permission of scope "id-token". available values are "read", "write" or "none" [permissions]
   |
24 |   id-token: rea
   |             ^~~
```

I'd expect an error to look like:
```log
.github/workflows/something.yaml:24:13: "read" is invalid for permission of scope "id-token". available values are "write" or "none" [permissions]
   |
24 |   id-token: read
   |             ^~~~
```
